lib/request.coffee

```diff
@@ -322,4 +306,53 @@ module.exports = getRequest = ({
 	# (or resolve to) a new response, or throw/reject.
 	###
 
+	###*
+	# @summary Refresh token on user request
+	# @function
+	# @public
+	#
+	# @description
+	# This function automatically refresh token on user request.
+	# The token is refreshed even if it is not expired yet.
+	# Error is thrown if user is not authorized.
+	#
+	# @param {String} options.url - relative url
+	#
+	# @returns {String} token - new token
+	#
+	# @example
+	# request.refreshToken
+	# 	baseUrl: 'https://api.resin.io'
+	###
+
+	exports.refreshToken = (options = {}) ->
+
+		{baseUrl} = options
+
+		# Only refresh if we have resin-auth
+		return if not (auth?)
+
+		exports.send
+			url: '/whoami'
+			baseUrl: baseUrl
+			refreshToken: false
+
+		# At this point we're sure there is a saved token,
+		# however the fact that /whoami returns 401 allows
+		# us to safely assume the token is expired
+
+		.catch
+			code: 'ResinRequestError'
+			statusCode: 401
+		, ->
+			console.log('error catched')
```

Comment:
Leftover console log

Reply:
Sorry for them. Keep missing those.

Reply:
Let me plug my magical solution to this problem: https://github.com/pimterry/git-confirm